### DISCUSSION TITLE
How do we handle prompt formatting?

### DISCUSSION THREAD

**Discussion by NeedsLoomis** [SciSharp/LLamaSharp#557] [Q&A]
Seems there are a couple ways to prompt AI's these days, depending on tuning and whatnot, for instance:

```
### Instruction:

### Response:

```
or

```
[INST] [/INST]
```

Do we simply pop them in our prompts? Does it interfere with the AuthorRole setting?

**Comment by patrick-hovsepian** [comment]
Hey, I'm currently working on something like this here https://github.com/SciSharp/LLamaSharp/pull/787 There's a set of models that llama.cpp supports internally and this PR looks to leverage those templates explicitly. However, it's not exhaustive of all models so you may still need to implement your own HistoryTransform depending on how the model expects the chat prompt

**Comment by n-ate** [comment]
Shouldn't prompt formatting depend on your LLM? There are several different prompt templats floating around on HuggingFace.
